index.js

```diff
@@ -78,6 +82,53 @@ class TrezorKeyring extends EventEmitter {
     return this.model;
   }
 
+  /**
+   * Gets the vendor, if known.
+   *
+   * @returns {"trezor" | "onekey"}
+   */
+  getVendor() {
+    return this.vendor;
+  }
```

Comment:
Getting closer I think. :)

With "lazy" I was meaning more like this:

```suggestion
  async getVendor() {
    return this.vendor || (this.vendor = await this._fetchVendor());
  }
```

, where `_fetchVendor` calls  `TrezorConnect.getFeatures()` and does the rest of what is currently in `__setVendor`.

The other new code further down would need to be changed as well to never call `__setVendor` explicitly.

Optionally you could inline everything that would go inside `fetchVendor()` here but I guess that's more a matter of preference.

Reply:
Sorry, I didn't understand the meaning of 'lazy' before.

Good suggestion, I will modify it like this. 🚀